HashRegistrar.sol

```diff
@@ -0,0 +1,340 @@
+    itself can be masqueraded by changing the required period or sending more than what you are 
+    bidding for. This is followed by a 24h reveal period. Bids revealed after this period will 
+    be burned and the ether unrecoverable. Since this is an auction, it is expected that most 
+    public hashes, like known domains and common dictionary words, will have multiple bidders pushing the price up. 
+    */ 
+    function newBid(bytes32 sealedBid) {
+        if (address(sealedBids[sealedBid]) > 0 ) throw;
+        // creates a new hash contract with the owner
+        Deed newBid = new Deed();
+        sealedBids[sealedBid] = newBid;
+        if (!newBid.send(msg.value)) throw;
+    } 
+    
+    
+    /*
+    ## Winning bids are deposited
+
+    The highest bid gets control of the hash, by depositing the equivalent amount of the second 
+    highest bid. The funds will be held on new contract, controlled only by the owner which 
+    contains basic information like when the hash was first registered, when it needs to be 
+    renewed, and the price paid for it. **The price paid is also saved on a moving average amount, 
+    calculated as ```averagePrice = averagePrice * 0.999 + newPrice * 0.001```**. The 
+    averagePrice at the moment of purchase is also registered on the contract.
+    */ 
+    function revealBid(bytes32 _hash, address _owner, uint _value, bytes32 _salt) noEther  {
+        bytes32 seal = shaBid(_hash, _owner, _value, _salt);
+        Deed bid = sealedBids[seal];
+        if (address(bid) == 0 ) throw;
+        sealedBids[seal] = Deed(0);
+        
+        entry h = entries[_hash];
+        
+        if (bid.creationDate() > h.auctionExpiration - revealPeriod
+            || now > h.auctionExpiration ) {
+            // bid is invalid, refund 99.9%
+            bid.closeDeed(1);
+            
+        } else if ( _value < averagePrice / minRatio ) {
+            // bid is invalid but not punishable
+            bid.closeDeed(1000);
+            
+        } else if (_value > h.highestBid) {
+            // new winner
+            // cancel the other bid, burn 0.1%
+            Deed previousWinner = h.deed;
+            previousWinner.closeDeed(999);
+            
+            // set new winner
+            h.value = h.highestBid;
+            h.highestBid = _value;
+            h.deed = sealedBids[seal];
+            bid.setOwner(_owner);
+            bid.setBalance(_value);
+        
+        } else if (_value > h.value) {
+            // not winner, but affects second place
+            h.value = _value;
+            bid.closeDeed(999);
+            
+        } else {
+            // bid doesn't affect auction
+            bid.closeDeed(999);
+        }
+    }
+    
+    function cancelBid(bytes32 seal) {
+        Deed bid = sealedBids[seal];
+        if (address(bid) == 0 || now < bid.creationDate() + auctionLength * 2 || bid.owner() > 0) throw; 
+        bid.closeDeed(0);
```

Comment:
Yes; this permits someone other than the bidder to delete bids for long-expired auctions. The delay is so that the legitimate bidder has an opportunity to reveal their bid and recover their ether.